sympy/core/basic.py

```diff
@@ -517,7 +522,7 @@ def atoms(self, *types):
         return result
 
     @property
-    def free_symbols(self) -> 'Set[Basic]':
+    def free_symbols(self) -> 'set[Symbol]':
```

Comment:
II don't think this is accurate given that
```python
In [17]: IndexedBase('I')[0].free_symbols
Out[17]: {I, I[0]}
```

Reply:
Thanks for pointing that out!

Reply:
Added to docstring as I may not be the only one that has this misunderstanding.